Twittimation/RandomLikeAutomaton.cs

```diff
@@ -0,0 +1,104 @@
+using Twittimation.Http;
+using Twittimation.IO;
+using Twittimation.Twitter;
+
+namespace Twittimation
+{
+    public class RandomLikeAutomaton : IAutomaton
+    {
+        private readonly IStored<RandomLikeAutomatonData> _data;
+        private readonly IStored<Tasks> _tasks;
+        private readonly Like _like;
+        private readonly IStored<Credentials> _credentials;
+        private int _updatesToSkip = 0;
+        private const int SecondsPerHour = 60 * 60;
+
+        public RandomLikeAutomaton(IStored<RandomLikeAutomatonData> data, IStored<Tasks> tasks, Like like, IStored<Credentials> credentials)
+        {
+            _data = data;
+            _tasks = tasks;
+            _like = like;
+            _credentials = credentials;
+        }
+
+        public void Update()
+        {
+            if(_updatesToSkip == 0)
+            {
+                _updatesToSkip = 11;
+                var credentials = _credentials.Get();
+                if (credentials.AreValid)
+                {
+                    var now = DateTimeOffset.UtcNow.ToUnixTimeSeconds();
+                    var random = new Random();
+                    try
+                    {
+                        var oauthClient = new OAuthClient("https://api.twitter.com/1.1/",
+                            credentials.ConsumerKey, credentials.ConsumerKeySecret, credentials.AccessToken, credentials.AccessTokenSecret);
+                        var followers = JObject.Parse(oauthClient.SendRequest("GET", "friends/ids.json", new JsonData()).Result);
+                        if (followers.ContainsKey("errors"))
+                        {
+                            Console.Error.WriteLine("Failed to get followers!");
+                            return;
+                        }
+                        var tasks = _tasks.Get();
+                        _data.Update(d => {
+                            if (d.LastUpdatedTimestamp < now - now % (SecondsPerHour * 24))
+                                d.LikesGivenPerFollowee = d.LikesGivenPerFollowee.ToDictionary(kvp => kvp.Key, kvp => 0);
+                            foreach(var id in followers.GetValue("ids").ToObject<long[]>())
+                            {
+                                var tweets = JsonConvert.DeserializeObject<List<TwitterTweet>>(
+                                    oauthClient.SendRequest("GET", "statuses/user_timeline.json",
+                                        new Dictionary<string, string>() {
+                                            { "user_id", id.ToString() },
+                                            { "count", "50" },
+                                            { "include_rts", "false" },
+                                            { "trim_user", "true" },
+                                            { "exclude_replies", "true" }
+                                        },
+                                        new JsonData())
+                                    .Result);
```

Comment:
Should never use `.Wait()` or `.Result` in production code. If you need to synchronize something, use `.GetAwaiter().GetResult()` instead.

Reply:
Those sound similar, what is the difference?

Reply:
There is a world of difference. This book is probably a good primer on the subject, if you want to learn the "why" and the vast array of different scenarios between those: https://www.amazon.com/gp/product/1449367569/ref=as_li_tl?ie=UTF8&camp=1789&creative=390957&creativeASIN=1449367569&linkCode=as2&tag=stepheclearys-20&linkId=VZYFOOYTQN57RD5L

There are also a lot of great blogs posts on it here: https://blog.stephencleary.com/

And there are many posts explaining it on StackOverflow, for the thousands of problems it has caused people.